sql/core/src/test/scala/org/apache/spark/sql/connector/AlterTableTests.scala

```diff
@@ -432,11 +432,23 @@ trait AlterTableTests extends SharedSparkSession with QueryErrorsBase {
     withTable(t) {
       sql(s"CREATE TABLE $t (id int) USING $v2Format")
 
-      val exc = intercept[AnalysisException] {
-        sql(s"ALTER TABLE $t ADD COLUMN point.z double")
-      }
-
-      assert(exc.getMessage.contains("Missing field point"))
+      val sqlText = s"ALTER TABLE $t ADD COLUMN point.z double"
```

Comment:
The error message obtained by executing this SQL is:

```
A column, variable, or function parameter with name `point` cannot be resolved. Did you mean one of the following in table `testcat`.`ns1`.`ns2`.`table_name`? [`id`]
```

 This seems a bit strange, is it suggesting that users should change to execute 

```
ALTER TABLE $t ADD COLUMN id.z double
```

Reply:
Here it is true that the parent column `point` does not exist, we give the user an error message that `point` filed does not exist, is it already clear to user?  If he insists on using `id.z`, he will get an more explicit `INVALID_FIELD_NAME`( Field name `id`.`z` is invalid: `id` is not a struct. SQLSTATE: 42000) error.